src/main/scala/edu/berkeley/ce/sparkrocks/SeedJointSelector.scala

```diff
@@ -3,242 +3,221 @@ package edu.berkeley.ce.sparkrocks
+    * @param remainingJoints Seq of joints that were not used in seed block generation
+    * @return Returns a tuple containing a Seq of blocks and a Seq of joints. The Seq of blocks
+    *         are the the desired partitions. The Seq of joints are the remaining joints that were
+    *         not used to generate the partitions.
     */
   @tailrec
-  private def findSeedJoints(jointSet: Seq[Joint], rockVolume: Block, numProcessors: Integer, tolerance: Double = 0.01,
-                             stepSize: Double = 0.01): Option[Seq[Joint]] = {
-    val totalVolume = rockVolume.volume
-    val volumePerProc = rockVolume.volume / numProcessors
-    val seedJoints = selectSeedJoints(jointSet, rockVolume, volumePerProc, totalVolume,
-      Seq.empty[Joint], tolerance, numProcessors)
-
-    if (stepSize < 0.01) {
-      // Unable to find satisfactory seed joints using current joint set
-      None
-    } else if (seedJoints.length != numProcessors - 1) {
-      val newStepSize = if (tolerance >= 1.0) { stepSize / 2.0 } else { stepSize }
-      val newTolerance = if (tolerance >= 1.0) { newStepSize } else { tolerance + stepSize }
-      findSeedJoints(jointSet, rockVolume, numProcessors, newTolerance, newStepSize)
-    } else {
-      val initialBlock = Seq(rockVolume)
-      val blocks = seedJoints.foldLeft(initialBlock) { (currentBlocks, joint) =>
+  def findSeedBlocks(jointSets: Seq[Seq[Joint]], inputVolume: Block, numPartitions: Int, jointSetSpan: Int,
+                     seedBlocks: Seq[Block] = Seq.empty[Block], remainingJoints: Seq[Joint] = Seq.empty[Joint]):
+  (Seq[Block], Seq[Joint]) = {
+    // All joint sets checked or span too large
+    if (jointSets.isEmpty || jointSetSpan > jointSets.length) {
+      (seedBlocks, remainingJoints)
+    } else if (jointSets.head(0).shape.isEmpty) {
+      val (seedJoints, leftOverJoints) = searchJointSets(jointSets, inputVolume, numPartitions, jointSetSpan)
+      val partitioning = seedJoints.foldLeft(Seq(inputVolume)) { (currentBlocks, joint) =>
         currentBlocks.flatMap(_.cut(joint))
       }
-      val nonRedundantBlocks = blocks map { case block @ Block(blockCenter, _, _) =>
-          Block(blockCenter, block.nonRedundantFaces)
+      // Insufficient partitions found
+      if (partitioning.length < numPartitions) {
+        findSeedBlocks(jointSets.tail, inputVolume, numPartitions, jointSetSpan, partitioning,
+          remainingJoints ++ jointSets.head)
+        // Sufficient partitions, return
+      } else {
+        (partitioning, remainingJoints ++ leftOverJoints)
       }
-      val volumes = nonRedundantBlocks.map(_.volume)
-      val avgVolume = volumes.sum / volumes.length
-      println("\nLOAD BALANCE INFORMATION:")
-      println(s"Average Volume: %.2f".format(avgVolume))
-      println(s"Max Volume: %.2f".format(volumes.max))
-      println(s"Min Volume: %.2f".format(volumes.min))
-      Some(seedJoints)
+      // Joint set non-persistent, check next one
```

Comment:
When you write a comment about conditionals, how about putting it on the first line inside the braces? That seems like it would make it more clear what the comment is actually about. I noticed this in a couple of other spots too.